woocommerce/woocommerce
# WC_Product_Variable_Data_Store_CPT: Validate that data exists in the transient

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

Currently we are validating the structure of transient data, however if we have the correct structure but the data is not present it will pass validation. This PR introduces a change to check that we have data present also during validation.

Closes https://github.com/woocommerce/woocommerce/issues/50855



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:


1. Install Transient Manager plugin and do a search for `wc_var_prices_` and `wc_product_children_` for the same product
2. Update the transient values with the corrupt data below
3. Visit the product page of this product and ensure it works as expected
4. Revisit these transients in the dashboard and check they have been update successfully with valid data.
5. Ensure unit tests are also passing

Corrupt `wc_product_children_` data:
```
a:3:{s:3:"all";a:0:{}s:7:"visible";a:0:{}s:7:"version";s:10:"1734130822";}
```

Corrupt `wc_var_prices_` data:
```
{"f9e544f77b7eac7add281ef28ca5559f":{"price":[],"regular_price":[],"sale_price":[]},"f74c5567e2b347f8728ea07774b6f7e4":{"price":[],"regular_price":
[],"sale_price":[]}}
```



### Changelog entry












<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
